generator/generator.py

```diff
@@ -1,31 +1,101 @@
-import csv
 import random
 
-def generator(shards, deltaT, N, mean, std):
-    tmp_list = []
+def generator(num_of_shards, delta_t, size_of_vectors, mean, std):
     vectors = []
+    tasks_in_delta_t = []
+    time_marks_in_delta_t = []
+    requests = []
 
-    for x in range(shards):
-        for y in range(N):
-            tmp_list.append(random.randint(0, 10))
-        vectors.append(tmp_list)
-        tmp_list = []
+    vectors = create_vectors(int(num_of_shards), int(size_of_vectors), float(mean), float(std))
 
+    save_vectors(vectors)
+            
+    tasks_in_delta_t = count_positive_tasks(int(size_of_vectors), vectors)
 
+    time_marks_in_delta_t = generate_time_stamps(int(size_of_vectors), float(delta_t), tasks_in_delta_t)
+    
+    requests = create_requests(int(size_of_vectors), tasks_in_delta_t, time_marks_in_delta_t, vectors)
+    
+    add_indexes_to_requests(requests)
+
+    save_requests(requests)
+
+def create_vectors(num_of_shards, size_of_vectors, mean, std):
+    vectors = []
+    for shard in range(num_of_shards): 
+        loads = []    
+        loads = np.random.normal(mean, std, size_of_vectors).tolist()
+        loads = [round(load, 3) for load in loads]
+        vectors.append(loads)
+    return vectors
+
+def save_vectors(vectors):
     vectors_file = open("./generator/vectors.csv","w")
-    for x in vectors:
-        vector_string = str(vectors.index(x)+1) +","+ ','.join(map(str, x)) + "\n"
-        vectors_file.write(vector_string)
+    for vector in vectors:
+       vector_string = str(vectors.index(vector)+1) +","+ ','.join(map(str, vector)) + "\n"
+       vectors_file.write(vector_string)
+    vectors_file.close()
+
+def count_positive_tasks(size_of_vectors, vectors):
+    tasks_in_delta_t = []
+    for column in range(size_of_vectors):
+        num_of_positive_loads = 0
+        for vector in vectors:
+            if(vector[column]>0):
+                num_of_positive_loads = num_of_positive_loads + 1
+        tasks_in_delta_t.append(num_of_positive_loads)
+    return tasks_in_delta_t
+
+def generate_time_stamps(size_of_vectors, delta_t, tasks_in_delta_t):
+    time_marks_in_delta_t = []
+    for size_of_vector in range(size_of_vectors):
+        delta_t_start = size_of_vector * delta_t
+        delta_t_end = (size_of_vector + 1) * delta_t
+        time_marks = np.random.uniform(delta_t_start, delta_t_end, tasks_in_delta_t[size_of_vector])
+        time_marks = [round(time_mark, 2) for time_mark in time_marks]
+        time_marks_in_delta_t.append(time_marks)
+    return time_marks_in_delta_t
+
+def create_requests(size_of_vectors, tasks_in_delta_t, time_marks_in_delta_t, vectors):
+    requests = []
+    for size_of_vector in range(size_of_vectors):
+        task_iterator = tasks_in_delta_t[size_of_vector] - 1
+        for vector in vectors:
+            if(vector[size_of_vector]>0):
+                tmp_req = []
+                tmp_req.append(time_marks_in_delta_t[size_of_vector][task_iterator])
```

Comment:
if we replace `task_iterator` with index of current vector wouldn't it be simpler and less code?

What I see is `size_of_vector` is column and `task_iterator` is row (maybe I'm missing something)

Reply:
`time_marks_in_delta_t` is a list of lists with timestamps for each period.  'task_iterator' is used to extract all timestamps from inner lists. 
The index of current vector in 'vectors' is the number of shard, so it cannot be used.

Reply:
you get inner list with `size_of_vector` and then you get an element by `task_iterator` so it's generally index of `vector` in `vectors`

Reply:
right, my bad, `task_iterator` and vector index are the same